kong/db/schema/entities/filter_chains.lua

```diff
@@ -18,22 +18,41 @@ local wasm = require "kong.runloop.wasm"
 
 ---@class kong.db.schema.entities.wasm_filter : table
 ---
----@field name    string
----@field enabled boolean
----@field config  string|table|nil
+---@field name       string
+---@field enabled    boolean
+---@field config     any|nil
+---@field raw_config string|nil
 
 
 local filter = {
   type = "record",
   fields = {
-    { name    = { type = "string", required = true, one_of = wasm.filter_names,
-                  err = "no such filter", }, },
-    { config  = { type = "string", required = false, }, },
-    { enabled = { type = "boolean", default = true, required = true, }, },
+    { name       = { type = "string", required = true, one_of = wasm.filter_names,
+                     err = "no such filter", }, },
+    { raw_config = { type = "string", required = false, }, },
+    { enabled    = { type = "boolean", default = true, required = true, }, },
+
+    { config = {
+        type = "json",
+        required = false,
+        json_schema = {
+          parent_subschema_key = "name",
+          namespace = constants.SCHEMA_NAMESPACES.PROXY_WASM_FILTERS,
+          optional = true,
+        },
+      },
+    },
```

Comment:
I'm thinking it might be better to reverse this, so `config` stays the same, and we add a new field called `json_config`.

1. It would simplify this PR, because we'd no longer require a migration to rename existing `config` fields to `raw_config`.
2. Users still reap the same convenience benefit of not needing to serialize their configuration, and `config`/`json_config` is arguably a bit more intuitive.
3. It's a little bit more forward-compatible on the off chance we want to support a different serialization format. Having a choice of `config | json_config | yaml_config` is more intuitive than `config | raw_config | yaml_config`

I don't think `config`/`raw_config` is _bad_ by any stretch, but if we want to go that route we can always do the `config` => `raw_config` / `json_config` => `config` swap in a smaller PR some time in the coming weeks.

@locao @hishamhm thoughts?

Reply:
BTW I have a commit for this change that is passing tests locally and ready to be pushed, so it won't add any extra turnaround time if we choose to do this. :)

Reply:
This feels like the right move for this PR; I'm making a judgment call and pushing that commit. As mentioned, I'm fine with re-introducing the change before the release if we'd like to.

Reply:
I think it's better for it to remain `config`, not `json_config`. At the level of our entity configuration, the meaning of `config` is that it is our nested configuration data structure, which happens to be validated via JSON Schema and which we happen to serialize as JSON when we pass to the filter. If we ever want to support different serialization formats, we can still use the same `config`, still validate it using JSON Schema, and then use a new field to specify the serialization format, instead of adding a new data-storage field.

As a side benefit, `config` helps pushing the JSON-based format as the "preferred default".

Reply:
Sounds reasonable enough to me; I don't have a strong opinion either way. If you feel like doing some more old school gateway development you can push this through while I'm out next week. Otherwise I don't think it should be too difficult for me to get it over the inline in time for feature freeze when I am back at my desk.